extensions/markdown/src/extension.ts

```diff
@@ -0,0 +1,170 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as vscode from 'vscode';
+import * as path from 'path';
+import { ExtensionContext, TextDocumentContentProvider, EventEmitter, Event, Uri, ViewColumn } from "vscode";
+
+const hljs = require('highlight.js');
+const mdnh = require('markdown-it-named-headers');
+const md = require('markdown-it')({
+	html: true,
+	highlight: function (str, lang) {
+		if (lang && hljs.getLanguage(lang)) {
+			try {
+				return `<pre class="hljs"><code><div>${hljs.highlight(lang, str, true).value}</div></code></pre>`;
+			} catch (error) { }
+		}
+		return `<pre class="hljs"><code><div>${md.utils.escapeHtml(str)}</div></code></pre>`;
+	}
+}).use(mdnh, {});
```

Comment:
When I say `on-the-fly`, it's more or less loading `anomynous` plugins:

```
plugins.each((ext) => {
  md.use(ext);
})
```
This is what we need to figure out (I have no idea about this either for now). 

Updating is another topic. One solution is keeping our hands clean by only allowing users to inject javascript files and let users update the plugins themselves. Otherwise we need careful design about plugin upgrade.